app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/BatteryNormalize.java

```diff
@@ -0,0 +1,44 @@
+package nodomain.freeyourgadget.gadgetbridge.util;
+
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+
+import nodomain.freeyourgadget.gadgetbridge.deviceevents.GBDeviceEventBatteryInfo;
+
+/**
+ * Created by Raziel on 13/04/2017.
+ */
+
+public class BatteryNormalize {
+    private int counter = 0;
+    private int levelMid = 0;
+    private GregorianCalendar lastCheck = new GregorianCalendar();
+    private static int INTERVAL_SECOND_CHECK  = 30;
+    private static int MAX_COUNT_CHECK = 6;
+
+    public void BatteryNormalize(int interval,int numerberCheck){
+        INTERVAL_SECOND_CHECK=interval;
+        MAX_COUNT_CHECK=numerberCheck;
+    }
+
+    //use this function if the battery level of the device oscillates
+    public short getLevelNormalize(int data, GBDeviceEventBatteryInfo cmd) {
```

Comment:
I'm not sure I completely understand how this class works, but it looks like you're basically just skipping some values. But how do you know that the value you're actually using is not one of the "wrong" values? Or do you just want to use *fewer* values to keep them from oscillating?

Did you think about using the median value instead?